### ISSUE TITLE
materialized view doesn't work if there is a "to" column

### ISSUE THREAD

**Issue by yaroot** [scylladb/scylla#9450]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash) :4.4.5-0.20211003.61469d62b-1
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu 20.04 (fresh install)

steps to reproduce

```
create table foo1
(
    id   int primary key,
    "to"    int
);

INSERT INTO foo1 (id, "to") values (1, 2);

create materialized view mv_foo1_to as
    select "to", "id" from foo1
    where "to" is not null and "id" is not null
primary key ( ("to", "id") );
```

error in logs:

```
[shard 1] view - Error executing build step for base test1.foo1: exceptions::syntax_exception (line 1:7 no viable alternative at input 'to')
```

**Comment by nyh** [comment]
A few more data points on this bug (I'm writing now a test to check all the angles):

Creating the view does not report any problem, the bug happens when **writing** data:
1. When writing to the base table which already has the view, the write fails with an unspecified "failure", and the following message appears in the log: `ERROR 2021-10-10 11:35:39,275 [shard 1] storage_proxy - exception during mutation write to ...: exceptions::syntax_exception (line 1:7 no viable alternative at input 'to')`
2. When the _view builder_ is writing the view for pre-existing data in the base table, the error is unfortunately more subtle - it is only logged as a warning (!) in the log: `WARN  2021-10-10 11:56:25,365 [shard 0] view - Error executing build step for base cql_test_1633856185345.cql_test_1633856185354: exceptions::syntax_exception (line 1:7 no viable alternative at input 'to')`

Additional notes:
* The bug does not require listing `"to"` on the `SELECT` of the view. A `SELECT *` reproduces the same bug.
* While `"to"` fails, `"dog"` does not. This is because the name `dog` can work unquoted, while `to` can't (it's a keyword (K_TO in Cql.g). 
* The column name `"Dog"` (note the quoted name with a capital letter) does work correctly. This suggests that perhaps our view-update code generates internal CQL which correctly quotes capitalized column names, but forgets that we need to quote all column names - e.g., in case they are keywords like "to".

**Comment by nyh** [comment]
Turns out the bug is in the required `"to" is not null` restriction.
What happens is that this restriction is parsed by the parser, into a restriction object, with the column name saved as the string `to` (without any quotes). Later the materialized-view filtering code wants to re-print the restriction it into a string. This **does not** re-add the quotes around the `to`, because it uses  `column_identifier::to_cql_string()` which uses `util::maybe_quote()`, which only adds quotes when the column name has non-lowercase characters - and does not add quotes around reserved keywords. So the view filtering code gets a restriction `to is not null` which cannot be parsed...

A trivial workaround is making `cql3::util::maybe_quote()` **always** decide to quote. Such a change fixs this bug, but I'm very worried that it might have additional implications - I don't remember why it was important for `maybe_quote` to **maybe** quote, and not always.

I'm still looking for other approaches. I don't remember why we need to parse the restriction and then re-print it as a string, instead of remembering the original string. Alternatively, `cql3::util::maybe_quote()` could have a blacklist of keywords that need quoting, and remember to quote those.

**Comment by nyh** [comment]
I just sent a patch to the mailing list titled "[PATCH] cql3: deprecate and start replacing cql3::util::maybe_quote", with a test and a fix for this issue.

**Comment by nyh** [comment]
This is a bug encountered multiple times by actual users (most recently, yesterday), so we should backport its fix.
The fix was already in branch 5.1.
Backported now to next-5.0 (9c24b78ae7726d93a7662d7e85d78962fb55e739) and next-4.6 (3f55ef10bfd623ed64944489bf6e82c1197a5344).
